session_test.go

```diff
@@ -109,22 +109,25 @@ func TestSessionServerClose(t *testing.T) {
 }
 
 func TestSessionCloseTimeout(t *testing.T) {
-	responder := func(remoteChannel uint16, req frames.FrameBody) ([]byte, error) {
+	responder := func(remoteChannel uint16, req frames.FrameBody) (fake.Response, error) {
 		switch req.(type) {
 		case *fake.AMQPProto:
-			return []byte{'A', 'M', 'Q', 'P', 0, 1, 0, 0}, nil
+			return newResponse(fake.ProtoHeader(fake.ProtoAMQP))
```

Comment:
Is ther some reason the fake.<functions> don't just return a Response object directly? Is there some use where you'd want to use them independent of a responder?

Reply:
That's what I originally had but after updating some of the tests it felt like it was mixing concerns.  e.g. `fake.ProtoHeader()` just creates a `[]byte` containing the marshalled frame and code like [this](https://github.com/Azure/go-amqp/blob/main/receiver_test.go#L254-L256) seems a bit awkward.